File: Lab1.2/Segment.cpp
#include "Segment.h"
#define _USE_MATH_DEFINES
#include <cmath>


Segment::Segment() {
    x1 = y1 = z1 = 1;
    x2 = y2 = z2 = 0;
}

Segment::Segment(double x1, double y1, double z1, double x2, double y2, double z2)
    : x1(x1), y1(y1), z1(z1), x2(x2), y2(y2), z2(z2) {
}

Segment::Segment(const Segment& other) {
    x1 = other.x1; y1 = other.y1; z1 = other.z1;
    x2 = other.x2; y2 = other.y2; z2 = other.z2;
}

Segment::~Segment() {
}

double Segment::getLength() const {
    return sqrt(pow(x2 - x1, 2) + pow(y2 - y1, 2) + pow(z2 - z1, 2));
}

double Segment::getAngleOY() const {
    double length = getLength();
    return acos(abs(y2 - y1) / length) * 180 / M_PI;
}

File: Lab 1.1.2/Segment.cs
using System;

public class Segment
{
    private double x1, y1, z1, x2, y2, z2;

    public Segment()
    {
        x1 = y1 = z1 = 0;
        x2 = y2 = z2 = 1;
    }

    public Segment(double x1, double y1, double z1, double x2, double y2, double z2)
    {
        this.x1 = x1;
        this.y1 = y1;
        this.z1 = z1;
        this.x2 = x2;
        this.y2 = y2;
        this.z2 = z2;
    }

    public Segment(Segment other)
    {
        x1 = other.x1;
        y1 = other.y1;
        z1 = other.z1;
        x2 = other.x2;
        y2 = other.y2;
        z2 = other.z2;
    }

~Segment()
    {
        Console.WriteLine("Segment object is being destroyed");
    }

    public double GetLength()
    {
        return Math.Sqrt(Math.Pow(x2 - x1, 2) + Math.Pow(y2 - y1, 2) + Math.Pow(z2 - z1, 2));
    }

    public double GetAngleOY()
    {
        double length = GetLength();
        return length == 0 ? 0 : Math.Acos(Math.Abs(y2 - y1) / length) * 180 / Math.PI;
    }
}
